spec/stack_master/parameter_resolvers/ami_finder_spec.rb

```diff
@@ -0,0 +1,69 @@
+RSpec.describe StackMaster::ParameterResolvers::AmiFinder do
+  subject(:resolver) { described_class.new('us-east-1') }
+  let(:ec2) { Aws::EC2::Client.new }
+
+  before do
+    allow(Aws::EC2::Client).to receive(:new).and_return(ec2)
+  end
+
+  describe '#build_filters' do
+    context 'when a single key-value pair is specified' do
+      it 'returns an array with a single hash' do
+        expect(resolver.build_filters('my-attr=my-value', nil)).to eq [
+          { name: 'my-attr', values: ['my-value']}
+        ]
+      end
+    end
+
+    context 'when multiple key-value pairs are specified' do
+      it 'returns an array with multiple hashes' do
+        expect(resolver.build_filters('my-attr=my-value,foo=bar', nil)).to eq [
+          { name: 'my-attr', values: ['my-value']},
+          { name: 'foo', values: ['bar']}
+        ]
+      end
+    end
+
+    context 'when a prefix is supplied' do
+      it 'adds the prefix to the filter' do
+        expect(resolver.build_filters('my-tag=my-value', 'tag')).to eq [
+          { name: 'tag:my-tag', values: ['my-value']}
+        ]
+      end
+    end
+
+    context 'owner_id is specified' do
+      it 'updates the owner instance variable' do
+        resolver.build_filters('owner_id=0123456789', nil)
+        expect(resolver.instance_variable_get('@owner')).to eq ['0123456789']
```

Comment:
It would be better to assert that the API was called with the expected value, rather than check the internal state of the object itself.